### PR TITLE
core: add tests for ReadableBuffers

### PR DESCRIPTION
**Pull request by jyane**
Address #4198, part 1. Just adding tests for `ReadableBuffers`.

### CODE REVIEW

**Comment by jyane** [comment]
@ejona86 Could you please review this? Might reviewer be @ carl-mastrangelo?

File: core/src/test/java/io/grpc/internal/ReadableBuffersTest.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  @Test
+  public void empty_returnsEmptyBuffer() {
+    ReadableBuffer buffer = ReadableBuffers.empty();
+    assertArrayEquals(new byte[0], buffer.array());
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void readArray_checksNotNull() {
+    ReadableBuffers.readArray(null);
+    fail("Exception expected");
```
Comment (ejona86) [inline core/src/test/java/io/grpc/internal/ReadableBuffersTest.java]: Fail is unnecessary. `expected` handles the case where no exception is thrown.

Comment (jyane) [reply core/src/test/java/io/grpc/internal/ReadableBuffersTest.java]: Done.

File: core/src/test/java/io/grpc/internal/ReadableBuffersTest.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      inputStream.read();
+    }
+    assertEquals(-1, inputStream.read());
+  }
+
+  @Test
+  public void bufferInputStream_read_returnsUnsignedByte() throws Exception {
+    ReadableBuffer buffer = ReadableBuffers.wrap(MSG_BYTES);
+    InputStream inputStream = ReadableBuffers.openStream(buffer, true);
+    // 104 represents 'h' in decimal ascii
```
Comment (dapengzhang0) [inline core/src/test/java/io/grpc/internal/ReadableBuffersTest.java]: 104 is not used in the code.

Comment (jyane) [reply core/src/test/java/io/grpc/internal/ReadableBuffersTest.java]: Removed.

**Comment by jyane** [comment]
I think I've addressed all comments :)

**Comment by ejona86** [comment]
Thanks!

**Pull request merged**